Sources/TuistCore/Utils/XCFrameworkMetadataProvider.swift

```diff
@@ -0,0 +1,62 @@
+import Basic
+import Foundation
+import TuistSupport
+
+enum XCFrameworkMetadataProviderError: FatalError, Equatable {
+    
+    case missingRequiredFile(AbsolutePath)
+
+    // MARK: - FatalError
+
+    var description: String {
+        switch self {
+        case let .missingRequiredFile(path):
+            return "The .xcframework at path \(path.pathString) doesn't contain an Info.plist. It's possible that the .xcframework was not generated properly or that got corrupted. Please, double check with the author of the framework."
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .missingRequiredFile:
+            return .abort
+        }
+    }
+}
+
+public protocol XCFrameworkMetadataProviding {
+    
+    /// Returns the available libraries for the xcframework at the given path.
+    /// - Parameter frameworkPath: Path to the xcframework.
+    func libraries(frameworkPath: AbsolutePath) throws -> [XCFrameworkInfoPlist.Library]
+    
+    /// Given a framework path and libraries it returns the path to its binary.
+    /// - Parameter frameworkPath: Framework path.
+    /// - Parameter libraries: Framework available libraries
+    func binaryPath(frameworkPath: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) -> AbsolutePath
+}
+
+public class XCFrameworkMetadataProvider: XCFrameworkMetadataProviding {
+
+    public func libraries(frameworkPath: AbsolutePath) throws -> [XCFrameworkInfoPlist.Library] {
+        let fileHandler = FileHandler.shared
+        let infoPlist = frameworkPath.appending(component: "Info.plist")
+        guard fileHandler.exists(infoPlist) else {
+            throw XCFrameworkMetadataProviderError.missingRequiredFile(infoPlist)
+        }
+        
+        let config: XCFrameworkInfoPlist = try fileHandler.readPlistFile(infoPlist)
+        return config.libraries
+    }
+    
+    public func binaryPath(frameworkPath: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) -> AbsolutePath {
+        let library = libraries.first { $0.architectures.contains(.arm64) }!
+        let binaryName = frameworkPath.basenameWithoutExt
+        let binaryPath =  AbsolutePath(library.identifier, relativeTo: frameworkPath)
+            .appending(RelativePath(library.path.pathString))
+            .appending(component: binaryName)
+
+        // Sanitising to make sure the precompiled binary located at binaryPath
+        _ = try! PrecompiledMetadataProvider().architectures(binaryPath: binaryPath).first!
```

Comment:
I'm still a bit fuzzy on this one :( - this would crash tuist if there are no architectures? is it intended to be a placeholder / todo?

As we saw xcframeworks may host various library types as such relying on `.arm64` variant to be present may not always work :/

Reply:
crash is not intended, ideally i would like to throw an error if the `xcframework` is missing architectures.

looking for `.arm64` only may not be enough, I would include check for other variants and return if any is found first.

Reply:
@pepibumur @kwridan as binaryPath is computed property, i cannot propagate an error from here, I could think of one alternative as follows

```suggestion
       override public var binaryPath: AbsolutePath {
        do {
            return try Self.metadataProvider.binaryPath(frameworkPath: path, libraries: libraries)
        }
        catch let error as XCFrameworkMetadataProviderError {
            Printer.shared.print(errorMessage: "\(error.description)")
            exit(-1)
        }
        catch let error as PrecompiledMetadataProviderError {
            Printer.shared.print(errorMessage: "\(error.description)")
            exit(-1)
        }
        catch _ {
            exit(-1)
        }
    }
```

or 

change computed property to a function to allow errors propagation and handle it uniformly from a single place

Reply:
@lakpa what you propose makes sense, it's a good point about where side effects can happen - if we'd like to keep the current approach then yes `binaryPath` would need to be changed to a throwing function as you suggested.

An alternative I'd like to put forward is, seeing we already parse the libraries when we first create the node and even throw errors when we don't find the `Info.plist`, perhaps we can also attempt to extract the primary binary then? Would that help at all or is that semantically incorrect?

Reply:
There are two things which could go wrong,

-  `Info.plist` is missing 

-  respective architecture references missing in the info.plist

Latter is covered, we could also implement covering former scenario